context_test.go

```diff
@@ -286,3 +287,23 @@ func TestContextWrapChild(t *testing.T) {
 		}
 	}
 }
+
+func TestContextInheritParentTimeout(t *testing.T) {
+	deadlineAfter := time.Now().Add(time.Hour)
+	pctx, cancel := context.WithTimeout(context.Background(), time.Hour)
+	defer cancel()
+
+	ctxBuilder := &ContextBuilder{
+		ParentContext: pctx,
+	}
+	ctx, cancel := ctxBuilder.Build()
+	defer cancel()
+
+	// Ensure deadline is in the future
+	deadline, ok := ctx.Deadline()
+	require.True(t, ok, "Missing deadline")
+
+	if deadline.Before(deadlineAfter) {
+		t.Fatalf("Expected deadline to be after %v, got %v", deadlineAfter, deadline)
```

Comment:
Optionally.
`assert.False(t, deadline.Before(deadlineAfter), "Expected deadline to be after %v, got %v", deadlineAfter, deadline)`?

Reply:
Done